Review 2

djpont opened the pull request:
Выполненные работы:
1. Переход на архитектуру Feature Sliced Design:
	- Рефакторинг кода и иерархии директорий.
	- Настройка ESLint для проверки импортов по правилам FSD.
	- Подготовили [документацию](https://github.com/agricultural-machinery-rental/client/blob/dev/docs/architecture.md) по архитектуре FSD.
2. Разработаны и обновлены базовые компоненты и сущности:
	- Компонент формы с использованием react-hook-form.
	- Обновлены компоненты (инпуты) под использование в react-hook-form.
	- Компонент выбора интервала дат.
	- Компонент чекбокс.
	- Компонент Textarea.
	- Компоненты заголовков.
	- Компонент InfoConditions (информация об условиях).
	- Компонент ProductCard  (карточка товара со слайдером).
3. Вёрстка страниц:
	- Главная страница с секцией "Категории техники".
	- О Компании.
4. Общие улучшения:
	- [Шаблон](https://github.com/agricultural-machinery-rental/client/blob/dev/.github/pull_request_template.md) для PR на GitHub.
	- Чистка package.json от неиспользуемых зависимостей, отказ от commitizen.

Краткий итог: реализованы основные сущности и базовые компоненты, произведена финальная инициализация проекта, улучшена командная работа благодаря утверждению архитектуры, правил для качественного кода. Подготовлен фундамент для дальнейшего масштабирования проекта.

Дальнейшие планы: создание страниц, подготовка MVP и документации.

djpont committed:
init: next, eslint, jest, prettier, workflow, storybook (#2)

Инициализация проекта

Lakatosska committed:
feat: add layout (#3)

Catalinadevolg committed:
feat: add styled-components (#4)

Catalinadevolg committed:
feat/006-header-footer (#6)

djpont committed:
#12445223 Routing (#7)

Add pages folders hierarchy
Eslint rule react/prop-types off.
Paths in utils.
Type for dynamic route pages.

djpont committed:
#12539089: Сброс глобальных стилей (#12)

Lakatosska committed:
Feat/005 UI (#8)

djpont committed:
#12594147: Чекбокс (#15)

Также добавил шрифт Open Sans и настроил Storybook для поддержки svg.
<img width="489" alt="image"
src="https://github.com/agricultural-machinery-rental/client/assets/34692754/815755d7-6981-4fc8-ad4a-385616e5e7c5">

djpont committed:
#12608236: Textarea (#16)

Компонент поля для ввода textarea.
<img width="383" alt="image"
src="https://github.com/agricultural-machinery-rental/client/assets/34692754/9cb8f95f-9e01-48e3-8699-ae20fc31bb6b">

djpont committed:
#12594147: Чекбокс (2) (#18)

djpont committed:
#12608236: Textarea (2) (#17)

djpont committed:
#12538889: Модальное окно (дизайн) (#10)

Как пользоваться:
1. В modalWindows (src/utils/modalWindows.tsx) добавляем в список
название нового содержимого окна.
```js
export enum modalWindows {
  example = 'example',
}
```
2. В modalWindowContent (src/utils/modalWindows.tsx) добавляем в
компонент нового содержимого окна.
```js
export const modalWindowContent = {
  [modalWindows.example]: <span>Компонент содержимого окна</span>,
} as const;
```
3. Импортируем
```js
import { ModalContext, modalWindows } from '@/providers/modalContext';
const { openModal } = useContext(ModalContext);
```
4. Вызываем с указанием modalWindows
```js
onClick={() => openModal(modalWindows.example)}
```

<img width="679" alt="image"
src="https://github.com/agricultural-machinery-rental/client/assets/34692754/51e42794-a0cf-4c52-b7d5-c80c969a672b">

Catalinadevolg committed:
#12512904: Вёрстка хэдера и футера (#13)

tltelf committed:
Верстка блока главной страницы (#20)

advatunes committed:
#12818828 компонент info conditions (информация об условиях)   (#21)

Co-authored-by: advatunes <dj-9-1-1@yandex.ru>

Catalinadevolg committed:
#12665371 Компонент формы (#19)

Co-authored-by: Alexander Votyakov <alexandervotyakov@gmail.com>

tltelf committed:
#12927680: Добавлены общие заголовки, произведен рефакторинг (#25)

1. Удалена лишняя обертка main

2. Произведен рефакторинг ChosenUs

3. Исправлены font-size и line-height в соответствии с макетом

4. В глобальных стилях создан общий заголовок h2, исправлены стили для
h3

5. В reset для всех заголовков добавлен шрифт и font-weight

6. В chosenUsData удален неиспользуемый id

Lakatosska committed:
#12896925 - Верстка секции "Категории техники" на главной странице (#24)

tltelf committed:
#13051469: Добавлен слайдер (#28)

tltelf committed:
#12898034 feat: add widget ProductCard (#29)

mrHoft committed:
feat: PR template, remove: commitizen (#33)

Catalinadevolg committed:
Рефакторинг компонента формы под FSD (#32)

Lakatosska committed:
#12933552 - split button functionality (#26)

в папку Features добавлены отдельные кнопки "Позвоните мне" и
"Забронировать", поправлены стили

advatunes committed:
#12911378 about company page (#30)

Co-authored-by: advatunes <dj-9-1-1@yandex.ru>
Co-authored-by: Alexander Votyakov <alexandervotyakov@gmail.com>

djpont committed:
#12911378 about company page (#35)

Co-authored-by: advatunes <dj-9-1-1@yandex.ru>

djpont committed:
#12821048: Компонент выбора интервала дат (#23)

Catalinadevolg committed:
Docs: add architecture doc file (#31)

djpont committed:
#13252141 Рефакторинг архитектуры FSD (#39)

Рефакторинг проекта под архитектуру FSD

Catalinadevolg committed:
#13021471: Форма авторизации (#40)

1. Task: Вёрстка формы авторизации

2. Screenshot: 

![image](https://github.com/agricultural-machinery-rental/client/assets/114286265/0e58961b-35e4-472b-b06a-f326abac83c3)


3. Was done:
   - [x] Вёрстка формы
   - [x] Поправить в форме отображение error
   - [x] Добавить возможность просмотра пароля

Lakatosska committed:
#12821053: Доработка хэдера (#42)

1. Task: Верстка хэдера в соответствии с утвержденным макетом

2. Was done:
   - [x] обновлен хэдер
   - [x] сделана кнопка-иконка трубка

Catalinadevolg committed:
#13266994: Окно восстановления пароля (#45)

1. Task: Вёрстка окна восстановления пароля 
Для теста окно открывается при нажатии на "Войти" в хэдере.

2. Screenshot:

![image](https://github.com/agricultural-machinery-rental/client/assets/114286265/21ee4087-dfc6-4700-a48c-7dbf70487bae)


3. Was done:
   - [x] Вёрстка

Lakatosska committed:
#13443559: Доработка футера (#46)

1. Task: Доделан хэдер, доработан футер

2. Screenshot:

![image](https://github.com/agricultural-machinery-rental/client/assets/89295707/37ff1784-3f4c-4fe7-a491-4e620afdce3e)


4. Was done:
   - [x] Доделан хэдер
   - [x] Заменен фон хэдера и логотип

Catalinadevolg committed:
#13021472: Окно регистрации (#47)

1. Task: Вёрстка окна регистрации
Для теста окно открывается при нажатии на "Войти" в хэдере.

2. Screenshot:

![image](https://github.com/agricultural-machinery-rental/client/assets/114286265/2ef001c3-65a2-4ab7-8e07-708ee0a18b7f)

3. Was done:
   - [x] Вёрстка

Catalinadevolg committed:
#13447906: Лэйаут страницы профиля (#43)

1. Task: Вёрстка меню на странице профиля

2. Screenshot:

![image](https://github.com/agricultural-machinery-rental/client/assets/114286265/58587563-bc7c-4056-82f2-997c6f2604bf)


3. Was done:
   - [x] Вёрстка меню

djpont committed:
#13121164 Вёрстка каталога (catalog item) (#48)

1. Task: Компонент CatalogItem, так же установил шрифт Ubuntu.

2. Примечание: Отличается символ `₽`

3. Что не сделано: Компонент "Добавить в избранное", жду реализации от
Кирилла.

4. Screenshot:
<img width="845" alt="image"
src="https://github.com/agricultural-machinery-rental/client/assets/34692754/551a9b5e-8fb7-4b28-adf0-e47728e4c48f">

---------

Co-authored-by: Nikolay Hoft <51874769+mrHoft@users.noreply.github.com>

MaxOrel commented:
1. Вижу что рефакторинг директорий произошел, но все должно быть единообразно, если вы называете папки компонентов/страниц с большой буквы то вы называете их всегда так.
![image](https://github.com/agricultural-machinery-rental/client/assets/27287886/32dc68ba-466e-4bc1-93f4-a59660777635)

MaxOrel commented:
2. Очень хорошо что есть документация по архитектуре.

MaxOrel commented on src/shared/styles/global.ts:
```diff
@@ -0,0 +1,77 @@
+});
+
+export const StContainer = styled.div`
+  width: 1104px;
```
Замените на max-width

MaxOrel commented on src/widgets/company/ChosenUs/styled.ts:
```diff
@@ -0,0 +1,18 @@
+
+export const StList = styled.ul`
+  display: grid;
+  grid-template-columns: repeat(3, 346px);
```
Используйте динамическое определение количества и размеров колонки

MaxOrel commented on src/widgets/company/ChosenUs/styled.ts:
```diff
@@ -0,0 +1,18 @@
+export const StList = styled.ul`
+  display: grid;
+  grid-template-columns: repeat(3, 346px);
+  grid-template-rows: repeat(2, 280px);
```
хотите задать высоту используйте grid-auto-rows: 280px; иначе блок не проходит тест на переполнение

MaxOrel reviewed (approved):
В целом замечания не значительные, работа выполнена отлично

advatunes committed:
#13252844 Хлебные крошки (#44)

1. Task: верстка Хлебные крошки

2. Screenshot:
![Screenshot 2023-08-10
132759](https://github.com/agricultural-machinery-rental/client/assets/63591058/cbe8eefb-0c6f-4570-a3d4-da055e37a186)


4. Was done:
   - [x] верстка Хлебные крошки (подстановка вручную через пропсы)
   - [x] подставление хлебных крошек из константы

---------

Co-authored-by: advatunes <dj-9-1-1@yandex.ru>
